indexing-service/src/main/java/io/druid/indexing/common/task/AbstractTask.java

```diff
@@ -104,6 +112,23 @@ public String getGroupId()
     return groupId;
   }
 
+  /**
+   * {@inheritDoc}
+   * */
+  @JsonIgnore
+  @Override
+  public int getLockPriority() {
+    return getContextValue("lockPriority") == null ? DEFAULT_TASK_PRIORITY :
+           Integer.parseInt(getContextValue("lockPriority").toString());
+  }
```

Comment:
can you define a static method getLockPriority(int defaultPriority) and use that here and in the classes that have different default priority?